src/workloads/query/InWithVariedArraySize.yml

```diff
@@ -0,0 +1,180 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query-execution"
+Description: |
+  This workload measures performance for $in for arrays of different sizes. We add the array length
+  for $in to the plan cache key since there is a possible explodeForSort optimization till we reach
+  internalQueryMaxScansToExplode that limits the maximum number of explode for sort index scans.
+  This workload measures performance for $in in both these cases, with and without indexes.
+Keywords:
+- in
+- cache
+- parametrization
+- classic
+- sbe
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  DocumentCount: &DocumentCount 1e6
+  MaxPhases: &MaxPhases 7
+  # This is internalQueryMaxScansToExplode - 3, since we append the generated array to [1, 2, 3] so
+  # we can actually match "int".
+  internalQueryMaxScansToExplode: &internalQueryMaxScansToExplode 197
+  InUptoMaxScansToExplode: &InUptoMaxScansToExplode {$in: {^Concat: {arrays: [[1, 2, 3], {^Array: {of: {^Inc: {start: 4}}, number: {^Inc: {start: 0, end: *internalQueryMaxScansToExplode}}}}]}}}
```

Comment:
In this query only 1, 2 and 3 are ever matched, so when we do an index scan, most of the index ranges will have no documents.

I would like that to change.
I saw several times (including the HELP ticket this is based on), that customers stuck a bunch of ids in $in. So let's go with that.

Let's use MonotonicLoader, so our documents will have _ids from 1 to 1e6.
Than we can generate $in arrays by just filling them with random integers from 1 to 1e6 and they will definitely match some documents.

This will make it similar to customer workload.

Reply:
Ah makes sense, now we will definitely match more docs. Done.